pkg/client/http.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Post performs a post request.
+func (c *PreflightClient) Post(path string, body io.Reader) (*http.Response, error) {
+	var bearer string
+	if !c.usingOAuth2() {
+		bearer = c.basicAuthToken
+	} else {
+		token, err := c.getValidAccessToken()
+		if err != nil {
+			return nil, err
+		}
+		bearer = token.bearer
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.fullURL(path), body)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	if len(bearer) > 0 {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", bearer))
+	}
+
+	return http.DefaultClient.Do(req)
+}
+
+func (c *PreflightClient) fullURL(path string) string {
+	base := c.baseURL
+	for strings.HasSuffix(base, "/") {
+		base = strings.TrimSuffix(base, "/")
+	}
+	for strings.HasPrefix(path, "/") {
+		path = strings.TrimPrefix(path, "/")
+	}
+	return fmt.Sprintf("%s/%s", base, path)
```

Comment:
can this use filepath join? like this? https://github.com/jetstack/preflight/pull/163/files#diff-fb87476fa66849c8514ba700e715b4b6R86

Reply:
this was only an idea so I've lgtm'd

shame prow doesn't have an option of /lgtm but let the author merge it... maybe it does...

Reply:
in this case, `filepath` does not work because the string contains `https://` and it assumes it is a file path, so the result ends up being `http:/domain/path` (with a single `/`).